nimble/host/include/host/ble_audio_codec.h

```diff
@@ -0,0 +1,201 @@
+/**
+ * @brief Remove codec entry from register
+ *
+ * @param[in] codec_record              Pointer to registered codec entry.
+ *
+ * @return                              0 on success;
+ *                                      A non-zero value on failure.
+ */
+int ble_audio_codec_unregister(struct ble_audio_codec_record *codec_record);
+
+/**
+ * @brief Iterate through all registered codecs and call function on every
+ *        one of them.
+ *
+ * @param[in] cb                        Callback to be called on codec entries.
+ * @param[in] arg                       Optional callback argument.
+ * @param[in] flags                     Codec entry flags. It is any
+ *                                      combination of following flags:
+ *                                          - BLE_AUDIO_CODEC_FLAG_SOURCE
+ *                                          - BLE_AUDIO_CODEC_FLAG_SINK
+ *                                      This filters entries so the callback is called
+ *                                      only on these that match the flag.
+ *
+ * @return                              0 on success;
+ *                                      A non-zero value on failure.
+ */
+int ble_audio_codec_foreach(uint8_t flags, ble_audio_codec_foreach_fn *cb, void *arg);
+
+typedef void ble_audio_codec_listener_fn(enum ble_audio_codec_event_op op,
+                                         struct ble_audio_codec_record *record, void *arg);
+
+/**
+ * Event listener structure
+ *
+ * This should be used as an opaque structure and not modified manually.
+ */
+struct ble_audio_codec_listener {
+    /** The function to call when a Codec event occurs. */
+    ble_audio_codec_listener_fn *fn;
+
+    /** An optional argument to pass to the event handler function. */
+    void *arg;
+
+    /** Singly-linked list entry. */
+    SLIST_ENTRY(ble_audio_codec_listener) link;
+};
+
+/**
+ * Registers listener for Codec events
+ *
+ * On success listener structure will be initialized automatically and does not
+ * need to be initialized prior to calling this function. To change callback
+ * and/or argument unregister listener first and register it again.
+ *
+ * @param listener      Listener structure
+ * @param fn            Callback function
+ * @param arg           Callback argument
+ *
+ * @return              0 on success
+ *                      BLE_HS_EINVAL if no callback is specified
+ *                      BLE_HS_EALREADY if listener is already registered
+ */
+int ble_audio_codec_listener_register(struct ble_audio_codec_listener *listener,
+                                      ble_audio_codec_listener_fn *fn, void *arg);
+/**
+ * Unregisters listener for Codec events
+ *
+ * @param listener      Listener structure
+ *
+ * @return              0 on success
+ *                      BLE_HS_ENOENT if listener was not registered
+ */
+int ble_audio_codec_listener_unregister(struct ble_audio_codec_listener *listener);
```

Comment:
I have mixed feelings defining dedicated listener for those.
Probably, audio listener can be used instead, introduced here: https://github.com/apache/mynewt-nimble/pull/1708

Reply:
Ok, #1708 looks like being close to merge so I'll just use that listener.